extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -181,88 +180,82 @@ export function combineInventoryUnits(instances) {
   };
 }
 
-/* eslint-disable jsdoc/require-param */
 /**
- * Array of functions used to combine block level request parameters for SRA
- * request.
- * @private @const {!Array<!function(!Array<AmpAdNetworkDoubleclickImpl>):?Object<string,string>>}
+ * Given array of instances, having previous called populateAdUrlState, join
+ * block state into SRA parameter.
+ * @enum {function(!Array<AmpAdNetworkDoubleclickImpl>):?Object<string,string>}
+ * @visibleForTesting
  */
-const BLOCK_SRA_COMBINERS_ = [
-  combineInventoryUnits,
-  // Although declared at a block-level, this is actually page level so
-  // return true if ANY indicate cookie opt out.
-  instances => getFirstInstanceValue_(instances, instance => {
-    return instance.jsonTargeting_ &&
-        instance.jsonTargeting_['cookieOptOut'] ? {'co': '1'} : null;
-  }),
-  instances => {
-    return {'adks': instances.map(instance => instance.adKey_).join()};
-  },
-  instances => {
-    return {'prev_iu_szs': instances.map(instance =>
-      instance.parameterSize_).join()};
-  },
-  // Although declared at a block-level, this is actually page level so
-  // return true if ANY indicate TFCD.
-  instances => getFirstInstanceValue_(instances, instance => {
-    return instance.jsonTargeting_ && instance.jsonTargeting_[TFCD] ?
-      {'tfcd': instance.jsonTargeting_[TFCD]} : null;
-  }),
-  // Although declared at a block-level, this is actually page level so
-  // return true if ANY indicate manual experiment.
-  instances => getFirstInstanceValue_(instances, instance => {
-    return isInManualExperiment(instance.element) ? {'adtest': 'on'} : null;
-  }),
-  instances => {
+export const SRA_JOINER = {
+  IU: combineInventoryUnits,
+  COOKIE_OPT_OUT: impls => getFirstInstanceValue_(impls, impl =>
```

Comment:
these all should have doc explaining what they do. While obviously I can read one of these arrow functions and understand what specifically it is actually doing, there is no background here on what or why anything is doing what it is doing